tests/unwind_safe.rs

```diff
@@ -20,12 +20,12 @@ fn writers_are_unwind_safe() {
     #[cfg(feature = "lzip")]
     {
         assert_unwind_safe::<lzma_rust2::LzipWriter<Vec<u8>>>();
-        assert_unwind_safe::<lzma_rust2::AutoFinishLzipWriter<Vec<u8>>>();
+        assert_unwind_safe::<lzma_rust2::AutoFinisher<lzma_rust2::LzipWriter<Vec<u8>>>>();
```

Comment:
Please also test that the other auto finisher are also unwind safe (making sure we don't break this contract in the future).